discovery/aws/ec2.go

```diff
@@ -185,8 +194,28 @@ func (d *EC2Discovery) ec2Client() (*ec2.EC2, error) {
 	return d.ec2, nil
 }
 
+func (d *EC2Discovery) azID(ctx context.Context, az string) (string, error) {
+	if azID, ok := d.azToAZID[az]; ok {
+		return azID, nil
+	}
+
+	azs, err := d.ec2.DescribeAvailabilityZonesWithContext(ctx, &ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return "", errors.Wrap(err, "could not describe availability zones")
+	}
+	d.azToAZID = make(map[string]string, len(azs.AvailabilityZones))
+	for _, az := range azs.AvailabilityZones {
+		d.azToAZID[*az.ZoneName] = *az.ZoneId
+	}
+
+	if azID, ok := d.azToAZID[az]; ok {
+		return azID, nil
+	}
+	return "", fmt.Errorf("no availability zone ID mapping found for %v", az)
```

Comment:
```suggestion
	return "", fmt.Errorf("no availability zone ID mapping found for %s", az)
```

Reply:
Changed, thanks. I've always used `%v` - is there a Go convention to prefer the precise verb for built-in types?

Reply:
yes, I think even go vet would complain

Reply:
It doesn't seem to. Not to worry.